Source/Diagnostics/ComputeDiagFunctors/JdispFunctor.cpp

```diff
@@ -0,0 +1,71 @@
+/* This file is part of Warpx.
+ *
+ * Authors: Avigdor Veksler
+ * License: BSD-3-Clause-LBNL
+*/
+#include "JdispFunctor.H"
+
+#include "WarpX.H"
+#include "FieldSolver/FiniteDifferenceSolver/HybridPICModel/HybridPICModel.H"
+#include "Particles/MultiParticleContainer.H"
+
+#include <AMReX.H>
+#include <AMReX_IntVect.H>
+#include <AMReX_MultiFab.H>
+
+JdispFunctor::JdispFunctor (int dir, int lev,
+        amrex::IntVect crse_ratio, bool convertRZmodes2cartesian, int ncomp)
+    : ComputeDiagFunctor(ncomp, crse_ratio), m_dir(dir), m_lev(lev),
+    m_convertRZmodes2cartesian(convertRZmodes2cartesian)
+{ }
+
+void
+JdispFunctor::operator() (amrex::MultiFab& mf_dst, int dcomp, const int /*i_buffer*/) const
+{
+    auto& warpx = WarpX::GetInstance();
+    auto* hybrid_pic_model = warpx.get_pointer_HybridPICModel();
+
+    /** pointer to source1 (total simulation current J) multifab */
+    amrex::MultiFab* m_mf_j = warpx.get_pointer_current_fp(m_lev, m_dir);
+    amrex::MultiFab* m_mf_j_ampere;
```

Comment:
I think we can be a bit more explicit here, rather than using the specific language of the hybrid-PIC model.
```suggestion
    amrex::MultiFab* m_mf_curlB;
```